inference/generativeai/llm-workshop/lab11-llama2/meta-llama-2-70b-lmi.ipynb

```diff
@@ -0,0 +1,1060 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "905903de-a4b9-4f41-8cc2-564416ae5d5f",
+   "metadata": {},
+   "source": [
+    "**Create the Tarball and then upload to S3 location**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c005aa2e-ec1a-4ccf-8f39-67bcbabd0309",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!rm model.tar.gz\n",
+    "!tar czvf model.tar.gz code_llama2_70b_fp16"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bb83ba3b-2ea5-4297-8e85-f16dd4c7c13a",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "s3_code_artifact = sess.upload_data(\"model.tar.gz\", bucket, s3_code_prefix)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "efa56b75-356a-4ebf-bebd-7150521c95e9",
+   "metadata": {},
+   "source": [
+    "### To create the end point the steps are:\n",
+    "\n",
+    "1. Create the Model using the Image container and the Model Tarball uploaded earlier\n",
+    "2. Create the endpoint config using the following key parameters\n",
+    "\n",
+    "    a) Instance Type is ml.g5.48xlarge \n",
+    "    \n",
+    "    b) ContainerStartupHealthCheckTimeoutInSeconds is 3600 to ensure health check starts after the model is ready    \n",
+    "3. Create the end point using the endpoint config created    \n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "ddee8273-8009-4739-af67-79525b6882cd",
+   "metadata": {},
+   "source": [
+    "#### Create the Model\n",
+    "Use the image URI for the DJL container and the s3 location to which the tarball was uploaded.\n",
+    "\n",
+    "The container downloads the model into the `/tmp` space on the instance because SageMaker maps the `/tmp` to the Amazon Elastic Block Store (Amazon EBS) volume that is mounted when we specify the endpoint creation parameter VolumeSizeInGB. It leverages `s5cmd`(https://github.com/peak/s5cmd) which offers a very fast download speed and hence extremely useful when downloading large models.\n",
+    "\n",
+    "For instances like p4dn, which come pre-built with the volume instance, we can continue to leverage the `/tmp` on the container. The size of this mount is large enough to hold the model.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eac40dc4-c9c3-4af1-8d54-7b6742259266",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from sagemaker.utils import name_from_base\n",
+    "\n",
+    "model_name = name_from_base(f\"Llama-2-70B-fp16-mpi\")\n",
+    "print(model_name)\n",
+    "\n",
+    "create_model_response = sm_client.create_model(\n",
+    "    ModelName=model_name,\n",
+    "    ExecutionRoleArn=role,\n",
+    "    PrimaryContainer={\"Image\": inference_image_uri, \"ModelDataUrl\": s3_code_artifact},\n",
```

Comment:
Could you please add the MODEL_LOADING_TIMEOUT environment variable? To avoid timeout when downloading the model.

```
    PrimaryContainer={"Image": inference_image_uri,
                      "ModelDataUrl": s3_code_artifact,
                      'Environment': {'MODEL_LOADING_TIMEOUT': '3600'}
                     }
```

Reply:
Updated the notebooks and added MODEL_LOADING_TIMEOUT